docs/dev_building_from_source.rst

```diff
@@ -0,0 +1,47 @@
+
+==============================
+Building libCellML from Source
+==============================
+
+This document covers building libCellML from source.  It is assumed that you already have the codebase downloaded and ready for building.  The variable LIBCELLML_SRC shall be used to refer to the directory containing the LICENSE file for libCellML.
+
+Build Directory
+===============
+
+It is best to build libCellML outside of the source tree.  To this end create a build directory that is not the LIBCELLML_SRC directory.  A sibling directory of LIBCELLML_SRC is a good choice, named something like::
+
+ * build
+ * libcellml-build
+
+Configuration
+=============
+
+The libCellML library uses the CMake build configuration tool to configure the library.  Version 3.2 or greater of CMake is required to configure libCellML.  Download and installation instructions are available from `CMake <http://www.cmake.org/>`_.
+
+The configuration options for the library are detailed in the following table.  The command line options can be set with the -D flag, like so -DBUILD_TYPE=Release.  Please note that in CMake GUI Configuration applications the config variable is prefixed with 'LIBCELLML\_'
```

Comment:
To make it easier to read, we might want to use ``` `` ``` to render things like ```-DBUILD_TYPE=Release```, i.e. use ``` ``-DBUILD_TYPE=Release`` ``` (and the same wherever else it would be useful).